std/traits.d

```diff
@@ -2809,7 +2809,7 @@ template FieldNameTuple(T)
     else static if (is(T == class))
         alias FieldNameTuple = staticMap!(NameOf, T.tupleof);
     else
-        alias FieldNameTuple = AliasSeq!"";
+        alias FieldNameTuple = AliasSeq!();
```

Comment:
But this is silently modifying behavior of code. The docs specifically say that "If `T` isn't a struct, class, or union, an expression tuple with an empty string is returned.". If we admit that an interface is not a class, then the current behavior is according to the spec.

If we decide that interfaces could be considered classes, then the fix should be made so that types that are not classes, unions or structs continue to return a tuple with an empty string.

Reply:
Interfaces  should be treated like classes because instances may have fields, they just aren't know at compile time. The current behaviour makes it pretty akward to enumerate fields using `static foreach (field; FieldNameTuple!T) ...` when `T` might be inferred as some interface.